huaweicloud/services/fgs/resource_huaweicloud_fgs_dependency_version.go

```diff
@@ -0,0 +1,264 @@
+				ForceNew:    true,
+				Description: "The region where the custom dependency version is located.",
+			},
+			"runtime": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The runtime of the custom dependency package and version.",
+			},
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the custom dependency package to which the version belongs.",
+			},
+			"link": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The OBS bucket path where the dependency package is located.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: "The description custom dependency version.",
+			},
+			"version": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The dependency package version.",
+			},
+			"version_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The ID of the dependency package version.",
+			},
+			"owner": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The dependency owner, public indicates a public dependency.",
+			},
+			"etag": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique ID of the dependency.",
+			},
+			"size": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: "The dependency size, in bytes.",
+			},
+		},
+	}
+}
+
+func buildDependencyVersionOpts(d *schema.ResourceData) dependencies.DependVersionOpts {
+	// Since the ZIP file upload is limited in size and requires encoding, only the OBS type is supported.
+	// The ZIP file uploading can also be achieved by uploading OBS objects and is more secure.
+	return dependencies.DependVersionOpts{
+		Name:        d.Get("name").(string),
+		Runtime:     d.Get("runtime").(string),
+		Description: utils.String(d.Get("description").(string)),
+		Type:        "obs",
+		Link:        d.Get("link").(string),
+	}
+}
+
+func resourceDependencyVersionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.FgsV2Client(cfg.GetRegion(d))
+	if err != nil {
+		return diag.Errorf("error creating FunctionGraph v2 client: %s", err)
+	}
+
+	resp, err := dependencies.CreateVersion(client, buildDependencyVersionOpts(d))
+	if err != nil {
+		return diag.Errorf("error creating custom dependency version: %s", err)
+	}
+	// Using depend ID and version number as the resource ID.
+	d.SetId(fmt.Sprintf("%s/%d", resp.DepId, resp.Version))
+
+	return resourceDependencyVersionRead(ctx, d, meta)
+}
+
+func ParseDependVersionResourceId(resourceId string) (dependId, versionInfo string, err error) {
+	parts := strings.Split(resourceId, "/")
+	if len(parts) < 2 {
```

Comment:
```suggestion
	if len(parts) != 2 {
```
if the parts len is greater than 2, it is error too

Reply:
if the length is greater than 2, the error will return in the query logic.